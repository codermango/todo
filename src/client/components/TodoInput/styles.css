.todoInput {
  margin: 10px;
  border: 1px solid #ededed;
  height: 35px;
}

.input {
  width: 100%;
  height: 100%;
  font-size: 16px;
}
