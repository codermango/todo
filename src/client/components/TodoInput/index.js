import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { addTodo } from '../../actions';

import styles from './styles.css';

class TodoInput extends Component {

  static propTypes = {
    addTodo: PropTypes.func.isRequired
  }

  state = {
    value: ''
  }

  handleKeyUp = (e) => {
    const { value } = this.state;
    if(e.keyCode === 13 && value) { // Enter
      this.props.addTodo({ text: this.state.value });
      this.setState({ value: '' });
    }
  }

  handleChange = (e) => {
    e.persist();
    const value = e.target.value;
    this.setState({ value });
  }

  render() {
    return (
      <div className={styles.todoInput}>
        <input
          className={styles.input}
          type="text"
          value={this.state.value}
          onChange={this.handleChange}
          onKeyUp={this.handleKeyUp}
          data-test-id="todo-input"
        />
      </div>
    )
  }
}

const mapDispatchToProps = dispatch => ({
  addTodo: ({ text }) => dispatch(addTodo({ text }))
});

export default connect(null, mapDispatchToProps)(TodoInput);
