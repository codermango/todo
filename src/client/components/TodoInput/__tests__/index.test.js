import React from 'react';
import { shallow } from 'enzyme';
import TodoInput from '../index';


describe('<TodoInput />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<TodoInput addTodo={jest.fn()} />);
    expect(wrapper.instance()).toMatchSnapshot();
  });

  it('handleChange works correctly', () => {
    const wrapper = shallow(<TodoInput addTodo={jest.fn()} />);
    const instance = wrapper.instance();
    const mockEvent = { persist: jest.fn(), target: { value: 'test' } };
    wrapper.find('input').simulate('change', mockEvent);
    expect(instance.state.value).toEqual('test');
  });

  describe('handleKeyUp works correctly', () => {
    const wrapper = shallow(<TodoInput addTodo={jest.fn()} />);
    const instance = wrapper.instance();

    beforeEach(() => {
      instance.props.addTodo.mockClear();
    });

    it('addTodo is called when user press Enter with value', () => {
      const mockEvent = { keyCode: 13 };
      wrapper.setState({ value: 'test' });
      wrapper.find('input').simulate('keyup', mockEvent);
      expect(instance.props.addTodo).toHaveBeenCalledTimes(1);
      expect(instance.state.value).toEqual('');
    });

    it('addTodo is not called when user press Enter without value', () => {
      const mockEvent = { keyCode: 13 };
      wrapper.find('input').simulate('keyup', mockEvent);
      expect(instance.props.addTodo).toHaveBeenCalledTimes(0);
    });

    it('addTodo is not called when user press other key with a value', () => {
      const mockEvent = { keyCode: 12 };
      wrapper.setState({ value: 'test' });
      wrapper.find('input').simulate('keyup', mockEvent);
      expect(instance.props.addTodo).toHaveBeenCalledTimes(0);
    });
  });
});
