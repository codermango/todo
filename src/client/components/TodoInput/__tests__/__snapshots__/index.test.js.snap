// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoInput /> renders correctly 1`] = `
TodoInput {
  "context": Object {},
  "handleChange": [Function],
  "handleKeyUp": [Function],
  "props": Object {
    "addTodo": [MockFunction],
  },
  "refs": Object {},
  "setState": [Function],
  "state": Object {
    "value": "",
  },
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useState": [Function],
      },
      "_element": <TodoInput
        addTodo={[MockFunction]}
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <div
        className="todoInput"
      >
        <input
          className="input"
          data-test-id="todo-input"
          onChange={[Function]}
          onKeyUp={[Function]}
          type="text"
          value=""
        />
      </div>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;
